### DISCUSSION TITLE
I'm a developer, not an end-user

### DISCUSSION THREAD

**Discussion by 4JX** [community/community#66334] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

You've got the flow of what information should be shown completely backwards with this change.

What I want to see when I land on the homepage is how *others* interact with *my* software, not what *I* may be interested in.

Seeing releases from projects I've starred, recommendations and co. is neat and all (the other discussions already do a good enough job of pointing out the issues with what counts as "following" a project) but provides little to no value to me as a developer most of the time.

What I consider actually valuable information is stuff like
- People starring or watching my repos
- Whether they are tinkering with it through forks
- New issues or pull requests that might need my attention (Which the "Recent activity" card gets right even if only for things I personally interact with beforehand)

Another pet peeve with the new feed is the apparent merging of "interactions that other users do with my projects that I own or otherwise actively participate in" and "events that happen on projects I follow", which just makes the value of the data provided at a glance much noisier than it needs to be.

**Comment by tvdijen** [comment]
Spot on @4JX !!

**Comment by roe-dl** [comment]
Very to the point.

**Comment by Xinayder** [comment]
Totally agree. If I want to see new repositories I'll go to the /explore page (which to be honest Microsoft already said they will retire that page). I don't want to see them on my feed. I want to see activities from people I follow on my feed, in CHRONOLOGICAL order. Not some AI bullshit.

**Comment by zjkal** [comment]
Totally agree

**Comment by nora-soderlund** [comment]
[One step forward, two steps back.](https://www.youtube.com/watch?v=lTHCwU9rUdY) I like the filter and stuff they've added - long waited for feature;

Buuut, they removed the whole reason to visit the feed, making the filter and the feed useless to most of us.

**Comment by LunarWatcher** [comment]
> Seeing releases from projects I've starred, [...] is neat and all (the other discussions already do a good enough job of pointing out the issues with what counts as "following" a project) but provides little to no value to me as a developer most of the time.

This is the only point I have to disagree with. You could already do that by watching the repo, and picking specifically releases. I've done that with a couple repos I want to keep up-to-date with, and not any others, because it's' generally not useful for me in the vast majority of cases.

See also https://docs.github.com/en/get-started/exploring-projects-on-github/saving-repositories-with-stars:

> Starring makes it easy to **find a repository or topic again later.** You can see all the repositories and topics you have starred by going to your [stars page](https://github.com/stars).

and https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications:

> When you watch a repository, **you're subscribing to updates for activity in that repository.**

What they've done is mix star and watch functionality, which blurs the currently clear line between what stars are for, and what watching is for, diminishing the usefulness and purpose of stars in the process. 

It's symptomatic of something else though: alongside the change to the feed and introduction of recommendation content, the amount of stuff included in the feed has increased drastically, and doing so has drastically increased the amount of noise in the feed. 

They've tried squeezing out as much content as they can from existing data sources to make the new feed changes look more valuable than it actually is, and that's a large part of the problem. All of the data additions to existing categories can't be toggled either, which means the filters can't be tweaked to restore the (at least rough) behaviour of the previous feed. Even if the same categories are included, the amount of extra content is so massive that it just buries all the stuff I care about.

I just disabled the feed, and I don't plan on re-enabling it unless they roll it back fast. Thanks to this change, I need to figure out some other way to get the information reliably for the future. (I'm aware of the script that reverts the change, but it isn't a permanent fix for now, and I'm more thinking about a solution that further filters the data into exclusively the things I'm interested in, which I might as well do if I _have_ to write code because GH screwed up the feed)

**Reply by 9001** [reply]
> I'm more thinking about a solution that further filters the data into exclusively the things I'm interested in, which I might as well do if I have to write code because GH screwed up the feed

i have been playing with this idea as well; a third-party website which lets you assemble your own feed, only including the exact parameters you're interested in (fork/star notifications for a list of repos, someone opening a new branch in a fork of your repo, and so on).

i suppose there's the drawback that it would result in fairly heavy traffic towards github to grab the info it needs, however this is a problem that github brought upon themselves, so it wouldn't be something to lose sleep over.

while i lack the bandwidth for another project currently, i hope someone else does something like this!

**Reply by 4JX** [reply]
> What they've done is mix star and watch functionality, which blurs the currently clear line between what stars are for, and what watching is for, diminishing the usefulness and purpose of stars in the process.

Yeah this is what I was getting at. I treat stars exclusively as "oh cool" and "might need this sometime" bookmarks that are saved for later.

That I star a project does not mean I want every possible loggable thing that it generates shoved into my feed, thats what watching and its filtering is for.

**Reply by LunarWatcher** [reply]
I don't really have the bandwidth either, but I used the old feed quite a bit, and losing it means a significant QOL drop, and makes keeping track of stuff so much harder that I might just have to reassign the resources to making an alternative

Based on the feedback being overwhelmingly negative so far though, I'm pretty sure many others are looking at the same option, and some of them are  bound to have capacity to build an alternative, which is why I'm personally holding off for now. I'm more or less gambling that someone with more bandwidth have already started with the code and/or planning

> i suppose there's the drawback that it would result in fairly heavy traffic towards github to grab the info it needs, however this is a problem that github brought upon themselves, so it wouldn't be something to lose sleep over.

Exactly, and if it was a problem, it would be reflected in the API quota limits. The API quotas remain quite generous IIRC (and as long as MS doesn't try to pull a Reddit on GitHub's API, that won't be a problem in the future either), and there's plenty of apps that source data from GitHub's API already, some of which fill missing functionality, or improve existing functionality

**Comment by RaphaelWoude** [comment]
This!

**Comment by chuongmep** [comment]
Keep it up !  I don't want GitHub become Gittok

**Comment by TriMoon** [comment]
Not to sound rude but...

Did anyone actually expected different stuff from a site owned by Micro$@$? :thinking: 
I have already moved my own repos to GitLab LONG ago, i suggest you all to do the same, that way we won't have any reason to visit this advert site anymore at all...
:vulcan_salute:

**Reply by HbmMods** [reply]
> that way we won't have any reason to visit this advert site anymore at all...

The reason I visit this site 90% of the time and why I use the feed at all is because I follow certain people whose commits I would like to see. Switching to Gitlab or whatever other site doesn't fix that.

**Reply by TriMoon** [reply]
People like you and their needs, are free to stay here :rofl: 
The rest is more interested in following software development, not individuals...

**Comment by hinell** [comment]
I would rather have capability to remove pull  requests I've opened, rather than that feed lol.

**Comment by alexhorner** [comment]
+1 I don't care that someone I follow decided to open a pull request on a project I have never heard of before, keep that separate for when I am bored and want to scroll around. I care about someone forking, starring or contributing to my own repositories in my own account or organisations I am in.

Why was this even changed? The two tabs provided the same amount of information in a better filtered way so you can choose what you care about. I NEVER used the first other-peopley tab and always switched to activity about my own stuff. That's all I care about, and now I can't even filter to specifically that using the filter options. Pull requests? Sure, have everyones, not just ones on your own repos... Useless.

**Reply by hinell** [reply]
@alexhorner When I learned that I can't remove PRs in my own projects, I was quite surprised. No wonder Github can't keep up with trashy PRs pushed all the time.

I would vote to replace `feed`™ with feature allowing me to remove them. Would be even better if could remove them elsewhere as well. It's the most basic feature.
